spacepy/datamodel.py

```diff
@@ -1174,13 +1165,24 @@ def SDcarryattrs(SDobject, hfile, path, allowed_attrs):
                 hfile[path].create_group(key)
                 toHDF5(hfile, SDobject[key], path=path+'/'+key, compression=h5_compr_type, compression_opts=h5_compr_opts)
             elif isinstance(value, allowed_elems[1]):
+                comptype, compopts = (None, None) if value.shape is ()\
```

Comment:
Checking whether the shape `is ()` raises a syntax warning (`"is" with a literal. Did you mean "=="?`)

If `value` is a scalar then `len(value.shape)` will be zero. If `value` is an empty array, then len still returns non-zero. Should this be checking the len (or similar)?

Reply:
It should be `==` not `is`...I'm not sure why I keep on doing that. Not looking for an empty array, looking for an array scalar, which will always have a shape of an empty tuple.